### PR TITLE
API-10179: make public license/groups properties more secure

### PR DESCRIPTION
**Pull request by grz-gajda**
# Deploy preview

https://624fed6409614e49456c137d--livechat-public-docs.netlify.app

# Links

Link your Jira ticket.

- [Jira](https://livechatinc.atlassian.net/browse/API-10179)

# Description

New scopes:
- `properties.license.value--my.rw`
- `properties.license.value--all.rw`
- `properties.group.value--my.rw`
- `properties.group.value--all.rw`

More details about accessing the methods `list/update/delete_license/group_properties`.

# Review and release

Apply changes and resolve conflicts. Once the described functionality lands on prod, let us know on #platform-docs-releases that your PR is ready for release. We will **merge** and **publish** the changes.

### CODE REVIEW

File: src/constant/index.js [hunk 1]
```diff
@@ -329,6 +329,34 @@ const SCOPES = [
...
+  {
+    id: "properties.license.value--all.rw",
+    description: "Read/write permission for managing license's properties (in all namespaces but with access only to their public properties)",
+    role: "administrator",
+    role_type: "secondary",
+    product: "Livechat"
+  },
+  {
+    id: "properties.group.value--my.rw",
+    description: "Read/write permission for managing groups' properties (only my namespace)",
```
Comment (ale-kacperczyk) [inline src/constant/index.js:348]:
```suggestion
    description: "Read/write permission for managing group properties (only my namespace)",
```

File: src/constant/index.js [hunk 2]
```diff
@@ -329,6 +329,34 @@ const SCOPES = [
...
+    role: "administrator",
+    role_type: "secondary",
+    product: "Livechat"
+  },
+  {
+    id: "properties.group.value--all.rw",
+    description: "Read/write permission for managing groups' properties (in all namespaces but with access only to their public properties)",
```
Comment (ale-kacperczyk) [inline src/constant/index.js:355]:
```suggestion
    description: "Read/write permission for managing group properties (in all namespaces but with access only to their public properties)",
```

File: src/pages/management/configuration-api/index.mdx [hunk 3]
```diff
@@ -2062,10 +2062,12 @@ Updates a property value within a license. This operation **doesn't** overwrite
...
 | -------------------- | --------------------------------------------------------------------------------- |
 | Method URL           | `https://api.livechatinc.com/v3.4/configuration/action/update_license_properties` |
-| Required scopes__*__ | -                                                                                 |
+| Required scopes      | `properties.license.value--my:rw`*, `properties.license.value--all:rw`**          |
 
-__*)__ If the requester's Client ID matches the namespace, setting license properties is always allowed.
-Otherwise, the permission depends on how the property was configured during [registration](#register-properties); parameter: `<property_name>.locations.<location>.access.<user>.write`.
+__*)__ If the requester's Client ID matches the namespace, setting license properties requires `properties.license.value--my:rw`.
+
+__**)__ If the requester's Client ID doesn't match the namespace, setting license properties requires `properties.license.value--all:rw`. Scope allows to
```
Comment (ale-kacperczyk) [inline src/pages/management/configuration-api/index.mdx:2069]:
```suggestion
__**)__ If the requester's Client ID doesn't match the namespace, setting license properties requires `properties.license.value--all:rw`. The scope allows to
```

File: src/pages/management/configuration-api/index.mdx [hunk 4]
```diff
@@ -2175,10 +2179,12 @@ Deletes the properties set within a license.
...
 | -------------------- | --------------------------------------------------------------------------------- |
 | Method URL           | `https://api.livechatinc.com/v3.4/configuration/action/delete_license_properties` |
-| Required scopes__*__ | -                                                                                 |
+| Required scopes      | `properties.license.value--my:rw`*, `properties.license.value--all:rw`**          |
+
+__*)__ If the requester's Client ID matches the namespace, deleting license properties requires `properties.license.value--my:rw`.
 
-__*)__ If the requester's Client ID matches the namespace, deleting license properties is always allowed.
-Otherwise, the permission depends on how the property was configured during [registration](#register-properties); parameter: `<property_name>.locations.<location>.access.<user>.write`.
+__**)__ If the requester's Client ID doesn't match the namespace, deleting license properties requires `properties.license.value--all:rw`. Scope allows to
```
Comment (ale-kacperczyk) [inline src/pages/management/configuration-api/index.mdx:2186]:
```suggestion
__**)__ If the requester's Client ID doesn't match the namespace, deleting license properties requires `properties.license.value--all:rw`. The scope allows to
```

File: src/pages/management/configuration-api/index.mdx [hunk 5]
```diff
@@ -2227,10 +2233,12 @@ Updates a property value within a group as the [property location](#property-loc
...
 | -------------------- | ------------------------------------------------------------------------------- |
 | Method URL           | `https://api.livechatinc.com/v3.4/configuration/action/update_group_properties` |
-| Required scopes__*__ | -                                                                               |
+| Required scopes      | `properties.group.value--my:rw`*, `properties.group.value--all:rw`**            |
 
-__*)__ If the requester's Client ID matches the namespace, setting group properties is always allowed.
-Otherwise, the permission depends on how the property was configured during [registration](#register-properties); parameter: `<property_name>.locations.<location>.access.<user>.write`.
+__*)__ If the requester's Client ID matches the namespace, setting group properties requires `properties.group.value--my:rw`.
+
+__**)__ If the requester's Client ID doesn't match the namespace, setting group properties requires `properties.group.value--all:rw`. Scope allows to
```
Comment (ale-kacperczyk) [inline src/pages/management/configuration-api/index.mdx:2240]:
```suggestion
__**)__ If the requester's Client ID doesn't match the namespace, setting group properties requires `properties.group.value--all:rw`. The scope allows to
```

File: src/pages/management/configuration-api/index.mdx [hunk 6]
```diff
@@ -2292,10 +2300,12 @@ For properties without an explicit value assignment, the method will return the
...
 | -------------------- | ----------------------------------------------------------------------------- |
 | Method URL           | `https://api.livechatinc.com/v3.4/configuration/action/list_group_properties` |
-| Required scopes__*__ | -                                                                             |
+| Required scopes      | `properties.group.value--my:rw`*, `properties.group.value--all:rw`**          |
+
+__*)__ If the requester's Client ID matches the namespace, retrieving group properties requires `properties.license.value--my:rw`.
 
-__*)__ If the requester's Client ID matches the namespace, retrieving group properties is always allowed.
-Otherwise, the permission depends on how the property was configured during [registration](#register-properties); parameter: `<property_name>.locations.<location>.access.<user>.read`.
+__**)__ If the requester's Client ID doesn't match the namespace, retrieving group properties requires `properties.license.value--all:rw`. Scope allows to
```
Comment (ale-kacperczyk) [inline src/pages/management/configuration-api/index.mdx:2307]:
```suggestion
__**)__ If the requester's Client ID doesn't match the namespace, retrieving group properties requires `properties.license.value--all:rw`.  The scope allows to
```

File: src/pages/management/configuration-api/index.mdx [hunk 7]
```diff
@@ -2344,10 +2354,12 @@ Deletes the properties set within a group.
...
 | -------------------- | ------------------------------------------------------------------------------- |
 | Method URL           | `https://api.livechatinc.com/v3.4/configuration/action/delete_group_properties` |
-| Required scopes__*__ | -                                                                               |
+| Required scopes      | `properties.group.value--my:rw`*, `properties.group.value--all:rw`**            |
+
+__*)__ If the requester's Client ID matches the namespace, deleting group properties requires `properties.group.value--my:rw`.
 
-__*)__ If the requester's Client ID matches the namespace, deleting group properties is always allowed.
-Otherwise, the permission depends on how the property was configured during [registration](#register-properties); parameter: `<property_name>.locations.<location>.access.<user>.write`.
+__**)__ If the requester's Client ID doesn't match the namespace, deleting group properties requires `properties.group.value--all:rw`. Scope allows to
```
Comment (ale-kacperczyk) [inline src/pages/management/configuration-api/index.mdx:2361]:
```suggestion
__**)__ If the requester's Client ID doesn't match the namespace, deleting group properties requires `properties.group.value--all:rw`. The scope allows to
```

File: src/pages/management/configuration-api/v3.3/index.mdx [hunk 8]
```diff
@@ -2065,10 +2065,12 @@ Updates a property value within a license. This operation **doesn't** overwrite
...
 | -------------------- | --------------------------------------------------------------------------------- |
 | Method URL           | `https://api.livechatinc.com/v3.3/configuration/action/update_license_properties` |
-| Required scopes__*__ | -                                                                                 |
+| Required scopes      | `properties.license.value--my:rw`*, `properties.license.value--all:rw`**          |
 
-__*)__ If the requester's Client ID matches the namespace, setting license properties is always allowed.
-Otherwise, the permission depends on how the property was configured during [registration](#register-properties); parameter: `<property_name>.locations.<location>.access.<user>.write`.
+__*)__ If the requester's Client ID matches the namespace, setting license properties requires `properties.license.value--my:rw`.
+
+__**)__ If the requester's Client ID doesn't match the namespace, setting license properties requires `properties.license.value--all:rw`. Scope allows to
```
Comment (ale-kacperczyk) [inline src/pages/management/configuration-api/v3.3/index.mdx:2072]:
```suggestion
__**)__ If the requester's Client ID doesn't match the namespace, setting license properties requires `properties.license.value--all:rw`. The scope allows to
```

File: src/pages/management/configuration-api/v3.3/index.mdx [hunk 9]
```diff
@@ -2128,10 +2130,12 @@ For properties without an explicit value assignment, the method will return the
...
 | -------------------- | ------------------------------------------------------------------------------- |
 | Method URL           | `https://api.livechatinc.com/v3.3/configuration/action/list_license_properties` |
-| Required scopes__*__ | -                                                                               |
+| Required scopes      | `properties.license.value--my:rw`*, `properties.license.value--all:rw`**        |
+
+__*)__ If the requester's Client ID matches the namespace, retrieving license properties requires `properties.license.value--my:rw`.
 
-__*)__ If the requester's Client ID matches the namespace, retrieving license properties is always allowed.
-Otherwise, the permission depends on how the property was configured during [registration](#register-properties); parameter: `<property_name>.locations.<location>.access.<user>.read`.
+__**)__ If the requester's Client ID doesn't match the namespace, retrieving license properties requires `properties.license.value--all:rw`. Scope allows to
```
Comment (ale-kacperczyk) [inline src/pages/management/configuration-api/v3.3/index.mdx:2137]:
```suggestion
__**)__ If the requester's Client ID doesn't match the namespace, retrieving license properties requires `properties.license.value--all:rw`. The scope allows to
```

File: src/pages/management/configuration-api/v3.3/index.mdx [hunk 10]
```diff
@@ -2178,10 +2182,12 @@ Deletes the properties set within a license.
...
 | -------------------- | --------------------------------------------------------------------------------- |
 | Method URL           | `https://api.livechatinc.com/v3.3/configuration/action/delete_license_properties` |
-| Required scopes__*__ | -                                                                                 |
+| Required scopes      | `properties.license.value--my:rw`*, `properties.license.value--all:rw`**          |
+
+__*)__ If the requester's Client ID matches the namespace, deleting license properties requires `properties.license.value--my:rw`.
 
-__*)__ If the requester's Client ID matches the namespace, deleting license properties is always allowed.
-Otherwise, the permission depends on how the property was configured during [registration](#register-properties); parameter: `<property_name>.locations.<location>.access.<user>.write`.
+__**)__ If the requester's Client ID doesn't match the namespace, deleting license properties requires `properties.license.value--all:rw`. Scope allows to
```
Comment (ale-kacperczyk) [inline src/pages/management/configuration-api/v3.3/index.mdx:2189]:
```suggestion
__**)__ If the requester's Client ID doesn't match the namespace, deleting license properties requires `properties.license.value--all:rw`. The scope allows to
```

File: src/pages/management/configuration-api/v3.3/index.mdx [hunk 11]
```diff
@@ -2230,10 +2236,12 @@ Updates a property value within a group as the [property location](#property-loc
...
 | -------------------- | ------------------------------------------------------------------------------- |
 | Method URL           | `https://api.livechatinc.com/v3.3/configuration/action/update_group_properties` |
-| Required scopes__*__ | -                                                                               |
+| Required scopes      | `properties.group.value--my:rw`*, `properties.group.value--all:rw`**            |
 
-__*)__ If the requester's Client ID matches the namespace, setting group properties is always allowed.
-Otherwise, the permission depends on how the property was configured during [registration](#register-properties); parameter: `<property_name>.locations.<location>.access.<user>.write`.
+__*)__ If the requester's Client ID matches the namespace, setting group properties requires `properties.group.value--my:rw`.
+
+__**)__ If the requester's Client ID doesn't match the namespace, setting group properties requires `properties.group.value--all:rw`. Scope allows to
```
Comment (ale-kacperczyk) [inline src/pages/management/configuration-api/v3.3/index.mdx:2243]:
```suggestion
__**)__ If the requester's Client ID doesn't match the namespace, setting group properties requires `properties.group.value--all:rw`. The scope allows to
```

File: src/pages/management/configuration-api/v3.3/index.mdx [hunk 12]
```diff
@@ -2295,10 +2303,12 @@ For properties without an explicit value assignment, the method will return the
...
 | -------------------- | ----------------------------------------------------------------------------- |
 | Method URL           | `https://api.livechatinc.com/v3.3/configuration/action/list_group_properties` |
-| Required scopes__*__ | -                                                                             |
+| Required scopes      | `properties.group.value--my:rw`*, `properties.group.value--all:rw`**          |
+
+__*)__ If the requester's Client ID matches the namespace, retrieving group properties requires `properties.group.value--my:rw`.
 
-__*)__ If the requester's Client ID matches the namespace, retrieving group properties is always allowed.
-Otherwise, the permission depends on how the property was configured during [registration](#register-properties); parameter: `<property_name>.locations.<location>.access.<user>.read`.
+__**)__ If the requester's Client ID doesn't match the namespace, retrieving group properties requires `properties.group.value--all:rw`. Scope allows to
```
Comment (ale-kacperczyk) [inline src/pages/management/configuration-api/v3.3/index.mdx:2310]:
```suggestion
__**)__ If the requester's Client ID doesn't match the namespace, retrieving group properties requires `properties.group.value--all:rw`. The scope allows to
```

File: src/pages/management/configuration-api/v3.3/index.mdx [hunk 13]
```diff
@@ -2347,10 +2357,12 @@ Deletes the properties set within a group.
...
 | -------------------- | ------------------------------------------------------------------------------- |
 | Method URL           | `https://api.livechatinc.com/v3.3/configuration/action/delete_group_properties` |
-| Required scopes__*__ | -                                                                               |
+| Required scopes      | `properties.group.value--my:rw`*, `properties.group.value--all:rw`**            |
+
+__*)__ If the requester's Client ID matches the namespace, deleting group properties requires `properties.group.value--my:rw`.
 
-__*)__ If the requester's Client ID matches the namespace, deleting group properties is always allowed.
-Otherwise, the permission depends on how the property was configured during [registration](#register-properties); parameter: `<property_name>.locations.<location>.access.<user>.write`.
+__**)__ If the requester's Client ID doesn't match the namespace, deleting group properties requires `properties.group.value--all:rw`. Scope allows to
```
Comment (ale-kacperczyk) [inline src/pages/management/configuration-api/v3.3/index.mdx:2364]:
```suggestion
__**)__ If the requester's Client ID doesn't match the namespace, deleting group properties requires `properties.group.value--all:rw`. The scope allows to
```

File: src/pages/management/configuration-api/v3.5/index.mdx [hunk 14]
```diff
@@ -2057,10 +2057,12 @@ Updates a property value within a license. This operation **doesn't** overwrite
...
 | -------------------- | --------------------------------------------------------------------------------- |
 | Method URL           | `https://api.livechatinc.com/v3.5/configuration/action/update_license_properties` |
-| Required scopes__*__ | -                                                                                 |
+| Required scopes      | `properties.license.value--my:ro`*, `properties.license.value--all:ro`**          |
 
-__*)__ If the requester's Client ID matches the namespace, setting license properties is always allowed.
-Otherwise, the permission depends on how the property was configured during [registration](#register-properties); parameter: `<property_name>.locations.<location>.access.<user>.write`.
+__*)__ If the requester's Client ID matches the namespace, setting license properties requires `properties.license.value--my:ro`.
+
+__**)__ If the requester's Client ID doesn't match the namespace, setting license properties requires `properties.license.value--all:ro`. Scope allows to
```
Comment (ale-kacperczyk) [inline src/pages/management/configuration-api/v3.5/index.mdx:2064]:
```suggestion
__**)__ If the requester's Client ID doesn't match the namespace, setting license properties requires `properties.license.value--all:ro`. The scope allows to
```

File: src/pages/management/configuration-api/v3.5/index.mdx [hunk 15]
```diff
@@ -2170,10 +2174,12 @@ Deletes the properties set within a license.
...
 | -------------------- | --------------------------------------------------------------------------------- |
 | Method URL           | `https://api.livechatinc.com/v3.5/configuration/action/delete_license_properties` |
-| Required scopes__*__ | -                                                                                 |
+| Required scopes      | `properties.license.value--my:ro`*, `properties.license.value--all:ro`**          |
+
+__*)__ If the requester's Client ID matches the namespace, deleting license properties requires `properties.license.value--my:ro`.
 
-__*)__ If the requester's Client ID matches the namespace, deleting license properties is always allowed.
-Otherwise, the permission depends on how the property was configured during [registration](#register-properties); parameter: `<property_name>.locations.<location>.access.<user>.write`.
+__**)__ If the requester's Client ID doesn't match the namespace, deleting license properties requires `properties.license.value--all:ro`. Scope allows to
```
Comment (ale-kacperczyk) [inline src/pages/management/configuration-api/v3.5/index.mdx:2181]:
```suggestion
__**)__ If the requester's Client ID doesn't match the namespace, deleting license properties requires `properties.license.value--all:ro`. The scope allows to
```

File: src/pages/management/configuration-api/v3.5/index.mdx [hunk 16]
```diff
@@ -2222,10 +2228,12 @@ Updates a property value within a group as the [property location](#property-loc
...
 | -------------------- | ------------------------------------------------------------------------------- |
 | Method URL           | `https://api.livechatinc.com/v3.5/configuration/action/update_group_properties` |
-| Required scopes__*__ | -                                                                               |
+| Required scopes      | `properties.group.value--my:rw`*, `properties.group.value--all:rw`**            |
 
-__*)__ If the requester's Client ID matches the namespace, setting group properties is always allowed.
-Otherwise, the permission depends on how the property was configured during [registration](#register-properties); parameter: `<property_name>.locations.<location>.access.<user>.write`.
+__*)__ If the requester's Client ID matches the namespace, setting group properties requires `properties.group.value--my:rw`.
+
+__**)__ If the requester's Client ID doesn't match the namespace, setting group properties requires `properties.group.value--all:rw`. Scope allows to
```
Comment (ale-kacperczyk) [inline src/pages/management/configuration-api/v3.5/index.mdx:2235]:
```suggestion
__**)__ If the requester's Client ID doesn't match the namespace, setting group properties requires `properties.group.value--all:rw`. The scope allows to
```

File: src/pages/management/configuration-api/v3.5/index.mdx [hunk 17]
```diff
@@ -2287,10 +2295,12 @@ For properties without an explicit value assignment, the method will return the
...
 | -------------------- | ----------------------------------------------------------------------------- |
 | Method URL           | `https://api.livechatinc.com/v3.5/configuration/action/list_group_properties` |
-| Required scopes__*__ | -                                                                             |
+| Required scopes      | `properties.group.value--my:ro`*, `properties.group.value--all:ro`**          |
+
+__*)__ If the requester's Client ID matches the namespace, retrieving group properties requires `properties.license.value--my:ro`.
 
-__*)__ If the requester's Client ID matches the namespace, retrieving group properties is always allowed.
-Otherwise, the permission depends on how the property was configured during [registration](#register-properties); parameter: `<property_name>.locations.<location>.access.<user>.read`.
+__**)__ If the requester's Client ID doesn't match the namespace, retrieving group properties requires `properties.license.value--all:ro`. Scope allows to
```
Comment (ale-kacperczyk) [inline src/pages/management/configuration-api/v3.5/index.mdx:2302]:
```suggestion
__**)__ If the requester's Client ID doesn't match the namespace, retrieving group properties requires `properties.license.value--all:ro`. The scope allows to
```

File: src/pages/management/configuration-api/v3.5/index.mdx [hunk 18]
```diff
@@ -2057,10 +2057,12 @@ Updates a property value within a license. This operation **doesn't** overwrite
 |                      |                                                                                   |
 | -------------------- | --------------------------------------------------------------------------------- |
 | Method URL           | `https://api.livechatinc.com/v3.5/configuration/action/update_license_properties` |
-| Required scopes__*__ | -                                                                                 |
+| Required scopes      | `properties.license.value--my:ro`*, `properties.license.value--all:ro`**          |
```
Comment (bkubiak) [inline src/pages/management/configuration-api/v3.5/index.mdx:2060]: are you sure about `ro` here? This method updates properties

File: src/pages/management/configuration-api/index.mdx [hunk 19]
```diff
@@ -2125,10 +2127,12 @@ For properties without an explicit value assignment, the method will return the
 |                      |                                                                                 |
 | -------------------- | ------------------------------------------------------------------------------- |
 | Method URL           | `https://api.livechatinc.com/v3.4/configuration/action/list_license_properties` |
-| Required scopes__*__ | -                                                                               |
+| Required scopes      | `properties.license.value--my:rw`*, `properties.license.value--all:rw`**        |
```
Comment (bkubiak) [inline src/pages/management/configuration-api/index.mdx:2130]: why do we require `rw` when the method just lists properties?

File: src/pages/authorization/scopes/index.mdx [hunk 20]
```diff
@@ -77,7 +77,7 @@ Scopes define access to certain resources. Your app will ask users for access to
 
 ## Properties scopes
 
-<Scopes scopes="properties--my:ro,properties--my:rw,properties--all:ro,properties--configuration:rw" />
+<Scopes scopes="properties--my:ro,properties--my:rw,properties--all:ro,properties.configuration:rw,properties.license.value--my.rw,properties.license.value--all.rw,properties.group.value--my.rw,properties.group.value--all.rw" />
```
Comment (bkubiak) [inline src/pages/authorization/scopes/index.mdx:80]: I think we can already remove old scopes `properties--my:ro` etc, if they are not used anymore in recent stable (v3.4+) version. Wdyt @ale-kacperczyk ?

**Review by bkubiak — COMMENTED** [review]
please check once again `ro` and `rw` in all updated methods v3.3-v3.5

File: src/pages/authorization/scopes/index.mdx (hunk 20 above)
Comment (ale-kacperczyk) [reply src/pages/authorization/scopes/index.mdx:80]: Yes, that's the approach we've adopted

File: src/constant/index.js [hunk 21]
```diff
@@ -329,6 +329,34 @@ const SCOPES = [
     role_type: "secondary",
     product: "LiveChat",
   },
+  {
+    id: "properties.license.value--my.rw",
+    description: "Read/write permission for managing license's properties (only my namespace)",
```
[the remark this answers is not in the document]
Comment (grz-gajda) [reply src/constant/index.js]: Applied.

File: src/constant/index.js [hunk 22]
```diff
@@ -329,6 +329,34 @@ const SCOPES = [
...
+    role: "administrator",
+    role_type: "secondary",
+    product: "Livechat"
+  },
+  {
+    id: "properties.license.value--all.rw",
+    description: "Read/write permission for managing license's properties (in all namespaces but with access only to their public properties)",
```
[the remark this answers is not in the document]
Comment (grz-gajda) [reply src/constant/index.js]: Applied.

File: src/constant/index.js (hunk 1 above)
Comment (grz-gajda) [reply src/constant/index.js]: Applied.

File: src/constant/index.js (hunk 2 above)
Comment (grz-gajda) [reply src/constant/index.js]: Applied.

File: src/pages/management/configuration-api/index.mdx (hunk 3 above)
Comment (grz-gajda) [reply src/pages/management/configuration-api/index.mdx]: Applied.

File: src/pages/management/configuration-api/index.mdx [hunk 23]
```diff
@@ -2125,10 +2127,12 @@ For properties without an explicit value assignment, the method will return the
...
+
+__*)__ If the requester's Client ID matches the namespace, retrieving license properties requires `properties.license.value--my:rw`.
 
-__*)__ If the requester's Client ID matches the namespace, retrieving license properties is always allowed.
-Otherwise, the permission depends on how the property was configured during [registration](#register-properties); parameter: `<property_name>.locations.<location>.access.<user>.read`.
+__**)__ If the requester's Client ID doesn't match the namespace, retrieving license properties requires `properties.license.value--all:rw`. Scope allows to
```
[the remark this answers is not in the document]
Comment (grz-gajda) [reply src/pages/management/configuration-api/index.mdx]: Applied.

File: src/pages/management/configuration-api/index.mdx (hunk 4 above)
Comment (grz-gajda) [reply src/pages/management/configuration-api/index.mdx]: Applied.

File: src/pages/management/configuration-api/index.mdx (hunk 6 above)
Comment (grz-gajda) [reply src/pages/management/configuration-api/index.mdx]: Applied.

File: src/pages/management/configuration-api/index.mdx (hunk 7 above)
Comment (grz-gajda) [reply src/pages/management/configuration-api/index.mdx]: Applied.

File: src/pages/management/configuration-api/v3.3/index.mdx (hunk 8 above)
Comment (grz-gajda) [reply src/pages/management/configuration-api/v3.3/index.mdx]: Applied.

File: src/pages/management/configuration-api/v3.3/index.mdx (hunk 10 above)
Comment (grz-gajda) [reply src/pages/management/configuration-api/v3.3/index.mdx]: Applied.

File: src/pages/management/configuration-api/index.mdx (hunk 19 above)
Comment (grz-gajda) [reply src/pages/management/configuration-api/index.mdx]: Replaced with `ro`.

File: src/pages/management/configuration-api/v3.5/index.mdx (hunk 18 above)
Comment (grz-gajda) [reply src/pages/management/configuration-api/v3.5/index.mdx]: Replaced with `rw`.

File: src/pages/management/configuration-api/v3.5/index.mdx [hunk 24]
```diff
@@ -2339,10 +2349,12 @@ Deletes the properties set within a group.
...
 | -------------------- | ------------------------------------------------------------------------------- |
 | Method URL           | `https://api.livechatinc.com/v3.5/configuration/action/delete_group_properties` |
-| Required scopes__*__ | -                                                                               |
+| Required scopes      | `properties.group.value--my:rw`*, `properties.group.value--all:rw`**            |
+
+__*)__ If the requester's Client ID matches the namespace, deleting group properties requires `properties.group.value--my:rw`.
 
-__*)__ If the requester's Client ID matches the namespace, deleting group properties is always allowed.
-Otherwise, the permission depends on how the property was configured during [registration](#register-properties); parameter: `<property_name>.locations.<location>.access.<user>.write`.
+__**)__ If the requester's Client ID doesn't match the namespace, deleting group properties requires `properties.group.value--all:rw`. Scope allows to
```
[the remark this answers is not in the document]
Comment (grz-gajda) [reply src/pages/management/configuration-api/v3.5/index.mdx]: Applied.

File: src/pages/management/configuration-api/v3.5/index.mdx (hunk 17 above)
Comment (grz-gajda) [reply src/pages/management/configuration-api/v3.5/index.mdx]: Applied.

File: src/pages/management/configuration-api/v3.5/index.mdx (hunk 16 above)
Comment (grz-gajda) [reply src/pages/management/configuration-api/v3.5/index.mdx]: Applied.

File: src/pages/management/configuration-api/v3.5/index.mdx (hunk 15 above)
Comment (grz-gajda) [reply src/pages/management/configuration-api/v3.5/index.mdx]: Applied.

File: src/pages/management/configuration-api/v3.5/index.mdx [hunk 25]
```diff
@@ -2120,10 +2122,12 @@ For properties without an explicit value assignment, the method will return the
...
 | -------------------- | ------------------------------------------------------------------------------- |
 | Method URL           | `https://api.livechatinc.com/v3.5/configuration/action/list_license_properties` |
-| Required scopes__*__ | -                                                                               |
+| Required scopes      | `properties.license.value--my:ro`*, `properties.license.value--all:ro`**        |
+
+__*)__ If the requester's Client ID matches the namespace, retrieving license properties requires `properties.license.value--my:ro`.
 
-__*)__ If the requester's Client ID matches the namespace, retrieving license properties is always allowed.
-Otherwise, the permission depends on how the property was configured during [registration](#register-properties); parameter: `<property_name>.locations.<location>.access.<user>.read`.
+__**)__ If the requester's Client ID doesn't match the namespace, retrieving license properties requires `properties.license.value--all:ro`. Scope allows to
```
[the remark this answers is not in the document]
Comment (grz-gajda) [reply src/pages/management/configuration-api/v3.5/index.mdx]: Applied.

File: src/pages/management/configuration-api/v3.3/index.mdx (hunk 11 above)
Comment (grz-gajda) [reply src/pages/management/configuration-api/v3.3/index.mdx]: Applied.

File: src/pages/management/configuration-api/v3.3/index.mdx (hunk 13 above)
Comment (grz-gajda) [reply src/pages/management/configuration-api/v3.3/index.mdx]: Applied.

File: src/pages/management/configuration-api/v3.5/index.mdx (hunk 14 above)
Comment (grz-gajda) [reply src/pages/management/configuration-api/v3.5/index.mdx]: Applied.

File: src/pages/management/configuration-api/index.mdx [hunk 26]
```diff
@@ -2062,10 +2062,12 @@ Updates a property value within a license. This operation **doesn't** overwrite
...
 | Method URL           | `https://api.livechatinc.com/v3.4/configuration/action/update_license_properties` |
-| Required scopes__*__ | -                                                                                 |
+| Required scopes      | `properties.license.value--my:rw`*, `properties.license.value--all:rw`**          |
 
-__*)__ If the requester's Client ID matches the namespace, setting license properties is always allowed.
-Otherwise, the permission depends on how the property was configured during [registration](#register-properties); parameter: `<property_name>.locations.<location>.access.<user>.write`.
+__*)__ If the requester's Client ID matches the namespace, setting license properties requires `properties.license.value--my:rw`.
+
+__**)__ If the requester's Client ID doesn't match the namespace, setting license properties requires `properties.license.value--all:rw`. Scope allows to
+set all properties from your namespace and [public properties](#register-property) from `core` and other namespaces.
```
[the remark this answers is not in the document]
Comment (grz-gajda) [reply src/pages/management/configuration-api/index.mdx:2073]: Mentioned there: https://github.com/livechat/livechat-public-docs/pull/1116/files#diff-efbf8548ecfe531985193e32410b5cba098f6f528ff7584f1d9655813b44aef1R1702

File: src/pages/management/configuration-api/v3.5/index.mdx [hunk 27]
```diff
@@ -2170,10 +2174,12 @@ Deletes the properties set within a license.
 |                      |                                                                                   |
 | -------------------- | --------------------------------------------------------------------------------- |
 | Method URL           | `https://api.livechatinc.com/v3.5/configuration/action/delete_license_properties` |
-| Required scopes__*__ | -                                                                                 |
+| Required scopes      | `properties.license.value--my:ro`*, `properties.license.value--all:ro`**          |
```
Comment (bkubiak) [inline src/pages/management/configuration-api/v3.5/index.mdx:2177]: rw 😄

File: src/pages/management/configuration-api/index.mdx [hunk 28]
```diff
@@ -1699,6 +1699,9 @@ Properties can be private or public.
 
 Properties are grouped in _namespaces_, which helps to distinguish which property belongs to a given Client Id. The property _namespace_ is named after the **Client Id** a property belongs to.
 
+> The exception of these rules is namespace named `core` which contains properties required to valid configuration
```
Comment (ale-kacperczyk) [inline src/pages/management/configuration-api/index.mdx:1702]:
```suggestion
```

File: src/pages/management/configuration-api/v3.5/index.mdx (hunk 27 above)
Comment (grz-gajda) [reply src/pages/management/configuration-api/v3.5/index.mdx]: 🙈

File: src/pages/management/configuration-api/index.mdx (hunk 28 above)
Comment (grz-gajda) [reply src/pages/management/configuration-api/index.mdx]: Removed.

File: src/pages/management/configuration-api/index.mdx [hunk 29]
```diff
@@ -1699,6 +1699,9 @@ Properties can be private or public.
...
+> of license (e.g. `max_chats_count`) and is preconfigured for every license by default.
```
[the remark this answers is not in the document]
Comment (grz-gajda) [reply src/pages/management/configuration-api/index.mdx]: Removed.

File: src/pages/management/configuration-api/index.mdx [hunk 30]
```diff
@@ -2227,10 +2236,12 @@ Updates a property value within a group as the [property location](#property-loc
...
 | Method URL           | `https://api.livechatinc.com/v3.4/configuration/action/update_group_properties` |
-| Required scopes__*__ | -                                                                               |
+| Required scopes      | `properties.group.value--my:rw`*, `properties.group.value--all:rw`**            |
 
-__*)__ If the requester's Client ID matches the namespace, setting group properties is always allowed.
-Otherwise, the permission depends on how the property was configured during [registration](#register-properties); parameter: `<property_name>.locations.<location>.access.<user>.write`.
+__*)__ If the requester's Client ID matches the namespace, setting group properties requires `properties.group.value--my:rw`.
+
+__**)__ If the requester's Client ID doesn't match the namespace, setting group properties requires `properties.group.value--all:rw`. The scope allows to
+set all properties from your namespace and [public properties](#register-property) from other namespaces.
```
[the remark this answers is not in the document]
Comment (grz-gajda) [reply src/pages/management/configuration-api/index.mdx]: Yeah, It'll be better. Replaced everywhere.

**Review by bkubiak — APPROVED** [review]

**Comment by ale-kacperczyk** [comment]
Could you rebase your branch, please?

**Pull request merged**